libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts

```diff
@@ -239,78 +233,109 @@ export class DelegationsIndexService {
     return new Set(types?.split(',').map((type) => type.trim()))
   }
 
-  /* Lookup delegations in index for user for specific scope */
+  /* Lookup delegations in index for user for specific scope(s) */
   async getDelegationRecords({
-    scope,
+    scopes,
     nationalId,
     direction = DelegationDirection.OUTGOING,
   }: {
-    scope: string
+    scopes: string[]
     nationalId: string
     direction: DelegationDirection
   }): Promise<PaginatedDelegationRecordDTO> {
-    const apiScope = await this.apiScopeModel.findOne({
+    // Validate that all scopes exist
+    const apiScopes = await this.apiScopeModel.findAll({
       where: {
-        name: scope,
+        name: { [Op.in]: scopes },
       },
     })
-    if (!apiScope) {
-      throw new BadRequestException('Invalid scope')
-    }
 
-    const delegationTypesSupportedByScope =
-      await this.apiScopeDelegationTypeModel
-        .findAll({
-          where: {
-            apiScopeName: apiScope.name,
-          },
-        })
-        .then((x) => x.map((d) => d.delegationType))
+    if (apiScopes.length !== scopes.length) {
+      const foundScopes = apiScopes.map((s) => s.name)
+      const invalidScopes = scopes.filter((s) => !foundScopes.includes(s))
+      throw new BadRequestException(
+        `Invalid scope(s): ${invalidScopes.join(', ')}`,
+      )
+    }
 
     if (!kennitala.isValid(nationalId)) {
       throw new BadRequestException('Invalid national id')
     }
 
-    const supportsCustom = delegationTypesSupportedByScope.includes(
-      AuthDelegationType.Custom,
-    )
-
     const where = {
       ...(direction === DelegationDirection.INCOMING
         ? { toNationalId: nationalId }
         : { fromNationalId: nationalId }),
       validTo: { [Op.or]: [{ [Op.gte]: new Date() }, { [Op.is]: null }] },
     }
 
-    const delegations = await this.delegationIndexModel
-      .findAll({
-        where: {
-          [Op.or]: [
-            {
-              ...where,
-              type: {
-                [Op.in]: delegationTypesSupportedByScope.filter(
-                  (d) => d !== AuthDelegationType.Custom,
-                ),
-              },
-            },
-            supportsCustom
-              ? {
-                  ...where,
-                  type: AuthDelegationType.Custom,
-                  customDelegationScopes: { [Op.contains]: [apiScope.name] },
-                }
-              : {},
-          ],
+    // Process each scope individually since different scopes might support different delegation types
+    const scopePromises = apiScopes.map(async (scope) => {
```

Comment:
This will increase the number of SQL queries a bit in the hot path as soon as we add more notificationScopes.

Can we include apiScopeDelegationType in the apiScope query above? 

Then assemble a unique list of delegationTypes and custom delegation scopes to pass into a single delegation index query? 

Maybe even make it a unique query?

Reply:
True, will refactor